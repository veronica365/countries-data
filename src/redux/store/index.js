import { configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import countryReducer from '../countries';

const store = configureStore({
  reducer: {
    countries: countryReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(logger),
  devTools: process.env.NODE_ENV !== 'production',
});

export default store;
