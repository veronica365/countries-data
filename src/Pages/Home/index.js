import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import Headline from '../../components/Headline';
import CountryList from '../../components/CountryList';
import { setCountryById } from '../../redux/countries';

export default function Home() {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(setCountryById({}));
  }, [dispatch]);

  return (
    <div className="home">
      <Headline />
      <CountryList id={false} />
    </div>
  );
}
