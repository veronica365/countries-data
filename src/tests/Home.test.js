import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Home from '../Pages/Home';
import store from '../redux/store';

describe('Home page', () => {
  it('should render without issues', () => {
    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
