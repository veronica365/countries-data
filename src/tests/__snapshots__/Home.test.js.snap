// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should render without issues 1`] = `
<div>
  <div
    class="home"
  >
    <div
      class="headline"
    >
      <div
        class="overlay"
      />
      <div
        class="info"
      >
        <h1>
          Country Data
        </h1>
        <p>
          Find out more about the countries of the world
        </p>
      </div>
    </div>
    <section
      class="countries"
    />
  </div>
</div>
`;
